### ISSUE TITLE
C7N Mailer - crashes when reporting lambda resources where runtime is blank

### ISSUE THREAD

**Issue by tbg-developers** [cloud-custodian/cloud-custodian#8330]
### Describe the bug

When reporting on lambda resources if it is reporting on an lambda that uses a package type of image it fails as there is no runtime attribute for it.

### What did you expect to happen?

C7N mailer should not rely on runtime being present and process the report as normal

### Cloud Provider

Amazon Web Services (AWS)

### Cloud Custodian version and dependency information

```shell
Custodian:   0.9.23
Python:      3.8.5 (tags/v3.8.5:580fbb0, Jul 20 2020, 15:57:54) [MSC v.1924 64 bit (AMD64)]
Platform:   win32
Using venv:  False
Docker: False
Installed:

argcomplete==2.0.0
attrs==22.2.0
boto3==1.26.72
botocore==1.29.72
docutils==0.17.1
importlib-metadata==4.13.0
importlib-resources==5.10.2
jmespath==1.0.1
jsonschema==4.17.3
pkgutil-resolve-name==1.3.10
pyrsistent==0.19.3
python-dateutil==2.8.2
pyyaml==6.0
s3transfer==0.6.0
six==1.16.0
tabulate==0.8.10
typing-extensions==4.5.0
urllib3==1.26.14
zipp==3.13.0
```

### Policy

```shell
vars:
  tag-filters: &tag-compliance-filters
    - "tag:environment": absent
    - not:
        - type: value
          key: "tag:environment"
          value_type: normalize
          value:
            [
              "production",
              "staging",
              "prod",
              "tooling",
              "tooling-1",
              "staging-1",
            ]
          op: in

policies:
  - name: policy002-prod-lambda-tag-compliance
    mode:
      type: periodic
      schedule: rate(1 day)
      role: arn:aws:iam::1234567890:role/policy002-prod-custodian-role
      tags:
        environment: prod
        owner: platform
        Name: custodian
    resource: lambda
    description: |
      Check lambdas are tagged correctly
    filters:
      - or: *tag-compliance-filters
    actions:
      - type: notify
        slack_template: slack_default
        slack_msg_color: warning
        violation_desc: No violation.
        action_desc: Policy002 - These lambdas have the incorrect tags on
        to:
          - https://hooks.slack.com/services/XXXXXX/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
        transport:
          type: sqs
          queue: https://sqs.eu-west-1.amazonaws.com/1234567890/prod-custodian-mailer-queue
```

### Relevant log/traceback output

```shell
`START RequestId: ba13b91e-993a-45d6-8b1c-8ab887d65db2 Version: $LATEST
[ERROR]	2023-02-28T11:56:32.493Z	ba13b91e-993a-45d6-8b1c-8ab887d65db2	Error starting mailer MailerSqsQueueProcessor(). 
Error: 'Runtime' 
Traceback (most recent call last):
File "/var/task/c7n_mailer/handle.py", line 33, in start_c7n_mailer
mailer_sqs_queue_processor.run(parallel)
File "/var/task/c7n_mailer/sqs_queue_processor.py", line 114, in run
self.process_sqs_message(sqs_message)
File "/var/task/c7n_mailer/sqs_queue_processor.py", line 147, in process_sqs_message
sns_delivery=True
File "/var/task/c7n_mailer/target.py", line 41, in handle_targets
slack_messages = slack_delivery.get_to_addrs_slack_messages_map(message)
File "/var/task/c7n_mailer/slack_delivery.py", line 56, in get_to_addrs_slack_messages_map
self.config['templates_folders'])
File "/var/task/c7n_mailer/utils.py", line 105, in get_rendered_jinja
region=sqs_message.get('region', ''))
File "/var/task/jinja2/environment.py", line 1301, in render
self.environment.handle_exception()
File "/var/task/jinja2/environment.py", line 936, in handle_exception
raise rewrite_traceback_stack(source=source)
File "msg-templates/slack_template.j2", line 11, in top-level template code
{{ format_resource(resource, policy['resource']) | replace('\\"', '"') | replace('"', '\\"') }}\n
File "/var/task/c7n_mailer/utils.py", line 387, in resource_format
resource['Runtime'])
KeyError: 'Runtime'
END RequestId: ba13b91e-993a-45d6-8b1c-8ab887d65db2
REPORT RequestId: ba13b91e-993a-45d6-8b1c-8ab887d65db2	Duration: 162.13 ms	Billed Duration: 163 ms	Memory Size: 1024 MB	Max Memory Used: 113 MB`
```

### Extra information or context

https://cloud-custodian.slack.com/archives/CAL4P6YE6/p1677585710899119

**Comment by ajkerrigan** [comment]
Thanks for the report :+1: . Addressing this in #8329

**Comment by ajkerrigan** [comment]
Closed via #8329
